packages/protocol/contracts/tokenvault/ERC20Vault.sol

```diff
@@ -295,7 +260,7 @@ contract ERC20Vault is
             // If it's a canonical token
             ERC20Upgradeable t = ERC20Upgradeable(token);
             ctoken = CanonicalERC20({
-                chainId: block.chainid,
+                chainId: uint64(block.chainid),
```

Comment:
- ` _encodeDestinationCall` doesn't seem like a great name for this function because it does a lot more then just encode the call data!
- Is the `isBridgedToken[token]` check necessary if we can just do `ctoken = bridgedToCanonical[token];` and check if it's 0 or not (totally not sure about this, would need to check more).
- Below the balance change it checked when doing the `transferFrom`. That's of course great but could use some comments that describes why it's done like that! (`transferFrom` can subtract a fee from the amount etc...)

Reply:
1. renamed `_encodeDestinationCall` to `_handleMessage`

Reply:
3. Done. comment added.

Reply:
2. I need sometime to check if isBridgedToken can be removed.